wurst/objects/items/ScrollsDefinition.wurst

```diff
@@ -0,0 +1,431 @@
+        ..setCooldownGroup(ABILITY_TSUNAMI.toRawCode())
+        ..setLumberCost(9)
+        ..setAbilities(ABILITY_TSUNAMI.toRawCode())
+        ..setStockReplenishInterval(120)
+        ..setModelUsed(LocalItems.scrollPurple)
+        ..setInterfaceIcon(Icons.bTNSnazzyScrollPurple)
+        ..setNameEnhance("Scroll of Tsunami")
+        ..setTooltipExtended(ST_TT)
+
+@compiletime function createCycloneScroll() returns ItemDefinition
+    return createUsableItem(ITEM_SCROLL_CYCLONE)
+        ..setCooldownGroup(ABILITY_CYCLONE.toRawCode())
+        ..setLumberCost(9)
+        ..setAbilities(ABILITY_CYCLONE.toRawCode())
+        ..setStockReplenishInterval(120)
+        ..setModelUsed(LocalItems.scrollCyan)
+        ..setInterfaceIcon(Icons.bTNBansheeMaster)
+        ..setNameEnhance("Scroll of Cyclone")
+        ..setTooltipExtended(SC_TT)
+
+
+// Cyclone Buff Ids, I believe Cyclone Extra is the buff applied to air unit, didn't confirm
+// Those buff are set in all cyclone-based abilities
+let BUFF_CYCLONE = 'Bcyc'
+let BUFF_CYCLONE_EXTRA = 'Bcy2'
+
+function onCyclone(unit target)
+    // Following code give the ability Ghost (visible) to the unit targeted by cyclone
+    // It allows other unit to walk through the target while it's tossed in the air
+    target.addAbility(GHOST_VIS_ID)
+
+    // periodically checking if the unit is still tossed in the air
+    doPeriodicallyTimed(0.1, 15) buffPoller ->
+        if not target.hasAbility(BUFF_CYCLONE) and not target.hasAbility(BUFF_CYCLONE_EXTRA)
+            target.removeAbility(GHOST_VIS_ID)
+            buffPoller.stop()
+
+function unit.isFire() returns bool
+    return this.getTypeId() == UNIT_FIRE
+        or this.getTypeId() == UNIT_MAGE_FIRE_SUMMONED
+        or this.getTypeId() == UNIT_MAGE_FIRE
+
+function onTsunami(unit caster, vec2 target)
+    // The range from the caster that the effect is centered.
+    let OFFSET_DISTANCE = 300.
+
+    // The range from the center for which units are affected.
+    let EFFECT_DISTANCE = 150.
+
+    // Compute the center of the extra damage.
+    let center = caster.getPos().polarOffset(caster.getPos().angleTo(target), OFFSET_DISTANCE)
+
+    // Enumerate all units that are in region for extra damage.
+    ENUM_GROUP.enumUnitsInRange(center, EFFECT_DISTANCE)
+
+    // Perform the extra damage for the matching units.
+    for u in ENUM_GROUP
+        // Only enemy structures are affected.
+        if u.isEnemyOf(caster) and IsUnitType(u, UNIT_TYPE_STRUCTURE)
+            // Destroy fires outright.
+            if u.isFire()
+                flashEffect(Objects.nagaDeath, u.getPos())
+                u.kill()
+            // Otherwise deal additional damage.
+            else
+                caster.damageTarget(u, TSUNAMI_STRUCT_DMG)
+
+function onRoot(unit caster, unit target)
+    if target.isTroll()
+        InstantDummyCaster.castTarget(caster.getOwner(), ABILITY_ROOT_TROLL, 1, Orders.ensnare, target)
+    else
+        new DummyCaster()
```

Comment:
Don't you need to set its lifespan?

Reply:
It's 5 by default, but I guess I should fix it to the root duration.